functorch/functorch/_src/aot_autograd.py

```diff
@@ -657,7 +658,13 @@ def functional_call(*args, **kwargs):
         with _stateless.reparametrize_module(
             mod, pytree.tree_unflatten(args[:params_len], params_spec)
         ):
-            out = mod(*args[params_len:], **kwargs)
+            if isinstance(mod, torch.fx.GraphModule):
```

Comment:
I won't block this, but it will be better if the call sites take care of interpreter-ing the GraphModule themselves. By doing so, they will ensure that this works even if you have a GraphModule and you wrap it in some dummy adapter function to massage the input/outputs. We do this in several cases. This logic won't work for that case.

Reply:
Even with this API, it is still relatively easy to accidentally do the wrong thing and lose stack traces. If we can't design the API so that this is impossible to happen (and given that I nuked your "change the compiled code" change, I don't think such an API design is possible), it is better to catch it through testing/linting and force people to think about it explicitly.